Automattic/jetpack
# JITM style enhancements

**Visual JITM adjustments**

Touching up a few style quirks that were found in a [previous PR](https://github.com/Automattic/jetpack/pull/7058).

- [x] JITMs need to properly clear and sit below "screen options" area (not next to it)
- [x] Alignment of `X` should align better with the down arrow of the "screen options" area and other dismiss buttons for wp-admin notices
- [x] Remove old JITM code (no longer needed)

**Before:**

![](https://cloud.githubusercontent.com/assets/1126811/26469912/53381e16-4161-11e7-8fc0-a208cabb3280.png)

![](https://cloud.githubusercontent.com/assets/22752103/26556081/71efd530-4466-11e7-922b-5ab09e1e285e.jpg)

![](https://cloud.githubusercontent.com/assets/22752103/26556082/71f7f26a-4466-11e7-8521-59f0eeab07ee.jpg)

![](https://cloud.githubusercontent.com/assets/1041600/26556758/d8675a5c-4472-11e7-9dab-d426a7e7cd60.png)



**After:**

* dismissal `x` aligned on most screen sizes. wp-admin has different breakpoints than calypso so I didn't want to bloat our code just to be 100% on every screen size. It's 99% though. 
* border widened to match wp-admin notice borders. 
* JITM position is fixed to sit below "screen options" area. 
* Button width: `flex-basis` adjusted so the button text is on one line (90% of the time anyway—hard to do one-size-fits all and keep it so flexible)

<img width="1265" alt="screen shot 2017-05-30 at 4 54 02 pm" src="https://cloud.githubusercontent.com/assets/214813/26604566/bd7f0c58-4558-11e7-8131-0ed7bdffa8b5.png">

<img width="1261" alt="screen shot 2017-05-30 at 4 55 23 pm" src="https://cloud.githubusercontent.com/assets/214813/26604651/0b747330-4559-11e7-930e-88cf7fd7716e.png">

**on smaller screens...**
<img width="803" alt="screen shot 2017-05-30 at 5 06 24 pm" src="https://cloud.githubusercontent.com/assets/214813/26605029/6542c5dc-455a-11e7-921b-125e215d0ba1.png">


<img width="699" alt="screen shot 2017-05-30 at 4 55 36 pm" src="https://cloud.githubusercontent.com/assets/214813/26604722/3c33d380-4559-11e7-8921-9340f3626f2d.png">


**To test**

1) Install akismet, but keep it deactivated. Visit `/wp-admin/edit-comments.php` and verify that you see a JITM for Akismet. It should go away if dismissed or Akismet is activated. 
2) Install VaultPress, but keep it deactivated. After publishing a post, verify that you see a JITM for VaultPress. 
2) Install VaultPress, but keep it deactivated. `Visit 1/wp-admin/update-core.php` and verify that you see a JITM for VaultPress. 
3) Install Woo Comm and activate it. Visit `/wp-admin/admin.php?page=wc-settings` and verify that you see a JITM for Woo Services plugin.
